CoWorkMobileApp/Controllers/SearchViewController.swift

```diff
@@ -6,26 +6,192 @@
 //
 
 import UIKit
+import Cartography
+import RealmSwift
 
-class SearchViewController: UIViewController {
-    
-    lazy var gradient: CAGradientLayer = {
-        let gradient = CAGradientLayer()
-        gradient.type = .axial
-        gradient.colors = [
-            ThemeColors.grayColor?.cgColor ?? UIColor.darkGray.cgColor,
-            UIColor.lightGray.cgColor,
-            UIColor.white.cgColor
-        ]
-        gradient.locations = [0, 0.25, 1]
-        return gradient
+class SearchViewController: UIViewController, UICollectionViewDelegate {
+    
+    private enum Section: Int, CaseIterable, CustomStringConvertible {
+        case allResults
+        
+        var description: String {
+            switch self {
+            case .allResults: return "All Results"
+            }
+        }
+    }
+    
+    struct WorkspaceItem: Hashable {
+        var workspaceId: String?
+    }
+    
+    private var collectionView: UICollectionView!
+    private var diffableDataSource: UICollectionViewDiffableDataSource<Section, WorkspaceItem>!
+    
+    private var workspaceListItems: [WorkspaceItem] = []
+    private var workspaceNotificationToken: NotificationToken?
+    private var workspaceResults: Results<Workspace>?
+    
+    let viewModel = SearchViewModel()
+    
+    let searchController: UISearchController = {
+        let searchController = UISearchController(searchResultsController: nil)
+        searchController.searchBar.placeholder = "Search for Workspaces"
+        searchController.searchBar.searchBarStyle = .default
+        searchController.searchBar.searchTextField.backgroundColor = .white
+        searchController.obscuresBackgroundDuringPresentation = false
+        return searchController
     }()
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        setupView()
+        configureDataSource()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        viewModel.getWorkspaceList()
```

Comment:
I would swap these two.  We want to make sure that our listeners are setup to listen to any changes on our realm query before the api call is made.